source/common/config/delta_subscription_state.cc

```diff
@@ -113,17 +118,19 @@ void DeltaSubscriptionState::handleBadResponse(const EnvoyException& e, UpdateAc
   ack.error_detail_.set_code(Grpc::Status::GrpcStatus::Internal);
   ack.error_detail_.set_message(e.what());
   disableInitFetchTimeoutTimer();
-  ENVOY_LOG(warn, "delta config for {} rejected: {}", type_url_, e.what());
-  callbacks_.onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason::UpdateRejected, &e);
+  ENVOY_LOG(warn, "delta config for {} rejected: {}", type_url(), e.what());
+  callbacks().onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason::UpdateRejected, &e);
 }
 
 void DeltaSubscriptionState::handleEstablishmentFailure() {
-  callbacks_.onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure,
-                                  nullptr);
+  callbacks().onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure,
+                                   nullptr);
 }
 
-envoy::api::v2::DeltaDiscoveryRequest DeltaSubscriptionState::getNextRequestAckless() {
-  envoy::api::v2::DeltaDiscoveryRequest request;
+envoy::api::v2::DeltaDiscoveryRequest* DeltaSubscriptionState::getNextRequestInternal() {
```

Comment:
You are transferring the ownership. Does unique_ptr<DeltaDiscoveryRequest> make more sense ?

Reply:
Not in this unusual case, because this is the thing that's getting passed around with the `void*` trick. If we were to return it as a typed unique_ptr, the caller would immediately just `release()` that. (And, can't have a sensible untyped unique_ptr).

Reply:
Hmm, well, on second thought, despite being ugly and clunky, maybe the unique_ptr and release() would actually function as some nice in-code documentation.

Reply:
Ah, ok, no, on third thought this is unfortunately mechanically impossible. NewGrpcMuxImpl has to receive these objects from both delta and SotW, so they need to be generic on DeltaDiscoveryRequest vs DiscoveryRequest, and so can't be unique_ptr, only void*.